### PR TITLE
testing/e2e: add credential pagination test

### PR DESCRIPTION
**Pull request by johanbrandhorst**

### CODE REVIEW

File: testing/internal/e2e/tests/base/paginate_credential_test.go [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+	sClient := scopes.NewClient(client)
+	cClient := credentials.NewClient(client)
+	newOrgId := boundary.CreateNewOrgApi(t, ctx, client)
+	t.Cleanup(func() {
+		ctx := context.Background()
+		_, err := sClient.Delete(ctx, newOrgId)
+		require.NoError(t, err)
+	})
+	newProjectId := boundary.CreateNewProjectApi(t, ctx, client, newOrgId)
+	newStoreId := boundary.CreateNewCredentialStoreStaticCli(t, ctx, newProjectId)
```
Comment (moduli) [inline testing/internal/e2e/tests/base/paginate_credential_test.go:143]: I think we want to use the method `CreateNewCredentialStoreStaticApi` instead?

File: testing/internal/e2e/tests/base/paginate_credential_test.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+	err = json.Unmarshal(output.Stdout, &initialCredentials)
+	require.NoError(t, err)
+
+	var returnedIds []string
+	for _, cred := range initialCredentials.Items {
+		returnedIds = append(returnedIds, cred.Id)
+	}
+
+	require.Len(t, initialCredentials.Items, c.MaxPageSize+1)
+	assert.Empty(t, cmp.Diff(returnedIds, credentialIds, cmpopts.SortSlices(func(i, j string) bool { return i < j })))
```
Comment (moduli) [inline testing/internal/e2e/tests/base/paginate_credential_test.go:81]: Do we want to add this to the targets test?

File: testing/internal/e2e/tests/base/paginate_credential_test.go [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+	assert.Empty(t, initialCredentials.RemovedIds)
+	assert.NotEmpty(t, initialCredentials.ListToken)
+	mapItems, ok := initialCredentials.GetResponse().Map["items"]
+	require.True(t, ok)
+	mapSliceItems, ok := mapItems.([]any)
+	require.True(t, ok)
+	assert.Len(t, mapSliceItems, c.MaxPageSize+1)
+
+	// Create a new credential and destroy one of the other Credentials
+	newCredentialId := boundary.CreateNewStaticCredentialPasswordCli(t, ctx, newStoreId, "user", "password")
```
Comment (moduli) [inline testing/internal/e2e/tests/base/paginate_credential_test.go:183]: I think we may want to use the Api method here, too?

**Review by mikemountain — APPROVED** [review]

File: testing/internal/e2e/tests/base/paginate_credential_test.go (hunk 1 above)
Comment (johanbrandhorst) [reply testing/internal/e2e/tests/base/paginate_credential_test.go:143]: Sure, can do!

File: testing/internal/e2e/tests/base/paginate_credential_test.go (hunk 2 above)
Comment (johanbrandhorst) [reply testing/internal/e2e/tests/base/paginate_credential_test.go:81]: Sure, why not :)

File: testing/internal/e2e/tests/base/paginate_credential_test.go (hunk 3 above)
Comment (johanbrandhorst) [reply testing/internal/e2e/tests/base/paginate_credential_test.go:183]: There isn't a helper for this. Generally, it doesn't matter to me how the setup/surrounding logic is done, as long as the method being tested is using the appropriate style (list, in this case). Does that make sense?

Comment (moduli) [reply testing/internal/e2e/tests/base/paginate_credential_test.go:183]: Yea. I follow your point. My main concern is being able to run this test standalone if needed, so either.
- we convert this to use an Api method
- or we add in a `boundary.AuthenticateAdminCli(t, ctx)` somewhere in this test.

Comment (johanbrandhorst) [reply testing/internal/e2e/tests/base/paginate_credential_test.go:183]: Great point. I'm happy to add an API helper for this.

Comment (johanbrandhorst) [reply testing/internal/e2e/tests/base/paginate_credential_test.go]: Done

**Review by moduli — APPROVED** [review]
Remind me... why does the `Cli` test take much longer to run compared to the Api test?
```
=== RUN   TestCliPaginateCredentials
    scope.go:60: Created Org Id: o_Wsy2NyR1R7
    scope.go:93: Created Project Id: p_NWBqp1kAFN
    credential.go:67: Created Credential Store: csst_3ZMe8T6EuG
    credential.go:114: Created Username/Password Credentials: credup_eSgeMmRRn6
--- PASS: TestCliPaginateCredentials (14.42s)
=== RUN   TestApiPaginateCredentials
    scope.go:25: Created Org Id: o_qhOGnKge0r
    scope.go:38: Created Project Id: p_bBN6ITeump
    credential.go:25: Created Credential Store: csst_dVepdvXdp8
    credential.go:152: Created Username/Password Credentials: credup_zFrkK3mtLm
--- PASS: TestApiPaginateCredentials (1.09s)
=== RUN   TestCliPaginateTargets
    scope.go:60: Created Org Id: o_5AfKTbg590
    scope.go:93: Created Project Id: p_JJykqZqNQz
    target.go:101: Created Target: ttcp_NUVtWcBASV
--- PASS: TestCliPaginateTargets (13.22s)
=== RUN   TestApiPaginateTargets
    scope.go:25: Created Org Id: o_waYyJlrqIN
    scope.go:38: Created Project Id: p_9jMbTKAtVp
--- PASS: TestApiPaginateTargets (1.00s)
```

**Pull request merged**